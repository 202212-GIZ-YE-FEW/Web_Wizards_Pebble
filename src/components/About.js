const About = () => {
    return (
        <section id='about'>
            <div
                className='
                    relative 
                    overflow-hidden 
                    bg-no-repeat 
                    bg-cover 
                    w-full
                    brightness-200
                    h-[231px]
                    md:h-[493px]
                    lg:h-[690px]
                    xl:h-[787px]
                    overlay-container
                '
                style={{
                    backgroundPosition: "center",
                    backgroundImage: "url(/images/about.png)",
                }}
            ></div>
            <div
                className='
                    absolute
                    inset-0
                    opacity-60
                    text-center
                    lg:relative
                '
            >
                <h2
                    className='
                        font-black
                        top-1/2
                        text-xl
                        md:text-4xl
                        lg:text-5xl
                        text-dark
                        mt-16
                        mb-4
                        md:mt-52
                    '
                >
                    About Pebble Works
                </h2>
                <p
                    className='
                        text-sm
                        relative
                        mx-auto
                        w-1/2
                        md:text-xl
                        lg:text-3xl
                    '
                >
                    Body placeholder for text paragraph. A paragraph is a
                    self-contained unit of text dealing with a particular point
                    or idea.
                </p>
            </div>
        </section>
    );
};

export default About;
