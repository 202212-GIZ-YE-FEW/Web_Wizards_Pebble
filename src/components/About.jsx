function Brief(props) {
    const { head, brief } = props;
    return (
        <div className='absolute inset-0 mx-auto translate-y-1/4 h-full w-full flex-col items-center justify-center lg:relative'>
            <h2
                className='
                m-1 text-center text-xl mdP:text-3xl font-bold md:m-4 md:text-5xl md:font-bold lg:text-5xl lg:font-bold'
            >
                {head}
            </h2>
            <p
                className='
                text-sm mr-12 ml-12 text-center md:ml-24 md:mr-24 md:text-2xl lg:ml-32 lg:mr-32 lg:text-2xl'
            >
                {brief}
            </p>
        </div>
    );
}

const About = (props) => {
    const { t } = props;

    return (
        <section id='about'>
            <div
                className='relative overflow-hidden  bg-no-repeat  bg-cover w-full top-[-40px] brightness-200 h-[214px] sm:h-[489px] lg:h-[689px] xl:h-[789px] 2xl:h-[889px] before:absolute before:inset-0 before:bg-zinc-50 before:opacity-10'
                style={{
                    backgroundPosition: "center",
                    backgroundImage: "url(/images/about.png)",
                }}
            ></div>
            <Brief head={t("about")} brief={t("brief")} />
        </section>
    );
};

export default About;
