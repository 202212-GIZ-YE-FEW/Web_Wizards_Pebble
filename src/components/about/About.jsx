import Image from "next/image";

function Brief(props) {
    const { head, brief } = props;
    return (
        <div className='absolute inset-0 lg:relative lg:my-28'>
            <div className='flex h-full w-full flex-col items-center justify-center'>
                <h2
                    className='
                    m-1 text-center text-xl font-black md:m-4 md:text-2xl lg:text-4xl font-sans'
                >
                    {head}
                </h2>
                <p
                    className='
                    text-md mx-12 text-center md:mx-24 md:text-2xl lg:mx-32 lg:text-2xl font-sans'
                >
                    {brief}
                </p>
            </div>
        </div>
    );
}

const About = (props) => {
    const { t } = props;

    return (
        <section id='about'>
            <div className='relative'>
                <Image
                    src='/images/about.png'
                    alt='About Page Image'
                    width={100}
                    height={100}
                    className='w-full -mt-10'
                    objectFit='contain'
                    layout='responsive'
                />
                <Brief head={t("about")} brief={t("brief")} />
            </div>
        </section>
    );
};

export default About;
