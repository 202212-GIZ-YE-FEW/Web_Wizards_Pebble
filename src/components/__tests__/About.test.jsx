import renderer from "react-test-renderer";
import About from "@/components/About";

it("renders correctly", () => {
    const tree = renderer.create(<About t={(str) => str} />).toJSON();
    expect(tree).toMatchSnapshot();
});
