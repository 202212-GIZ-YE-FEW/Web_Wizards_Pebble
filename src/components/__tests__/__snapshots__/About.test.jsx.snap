// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  id="about"
>
  <div
    className="   relative   overflow-hidden    bg-no-repeat    bg-cover    w-full   top-[-40px]   brightness-200   h-[214px]   sm:h-[489px]   lg:h-[689px]   xl:h-[789px]   2xl:h-[889px]   before:absolute   before:inset-0   before:bg-stone-400   before:opacity-20   "
    style={
      Object {
        "backgroundImage": "url(/images/about.png)",
        "backgroundPosition": "center",
      }
    }
  />
  <div
    className="absolute   inset-0   w-[250px]   mx-auto   my-[40px]   translate-y-1/2   text-center   sm:w-[450px]   lg:relative   lg:w-[650px]   "
  >
    <h2
      className="   relative   font-black   text-black   mb-4   sm:text-3xl   lg:text-5xl   "
    >
      about
    </h2>
    <p
      className="   relative   mx-auto   sm:text-2xl   lg:text-4xl   leading-relaxed   sm:leading-relaxed   text-body-color   "
    >
      briefLineOne
       
      <br />
       
      briefLineTwo
    </p>
  </div>
</section>
`;
